tests/Doctrine/Tests/ORM/Query/SelectSqlMappingTest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Query;
+
+use Doctrine\DBAL\Types\FloatType;
+use Doctrine\DBAL\Types\IntegerType;
+use Doctrine\ORM\Query as ORMQuery;
+use Doctrine\Tests\OrmTestCase;
+use ReflectionException;
+use ReflectionObject;
+use function array_search;
+use function sprintf;
+
+class SelectSqlMappingTest extends OrmTestCase
+{
+    private $em;
+
+    protected function setUp(): void
+    {
+        $this->em = $this->getTestEntityManager();
+    }
+
+    /**
+     * Assert a valid SQL generation.
+     *
+     * @throws ReflectionException
+     */
+    public function assertScalarTypes(string $dqlToBeTested, array $types)
+    {
+        $query = $this->em->createQuery($dqlToBeTested);
+        $query
+            ->setHint(ORMQuery::HINT_FORCE_PARTIAL_LOAD, true)
+            ->useQueryCache(false);
+        $r = new ReflectionObject($query);
+        $method = $r->getMethod('getResultSetMapping');
+        $method->setAccessible(true);
```

Comment:
I think that we should turn this test into a functional one, avoiding the use of reflection - which makes the test somewhat fragile.

Reply:
I don't know how to solve this test. I can't find another entry point to reach the resultSetMapping to check if everything is working as expected.

Reply:
Don't worry, we can take care of that. Thanks a lot for your help!